### PR TITLE
Make additional announcable cidrs configurable per tenant super network

### PR DESCRIPTION
**Pull request by majst01**
Actually we have the Pod- and Service- Cidrs hard coded in metal-core and add them to the route maps per machine switch port during the creation of the frr.conf on a switch.

With this we put this as a optional configuration into the super tenant network called `additionalannouncablecidrs` which can contain a slice of cidrs which should be added to the route map of every machine created in his tenant network.

In https://github.com/metal-stack/metal-core/pull/127 we already filter out duplicate entries in the route maps to keep backwards compatibility.

currently running in Test

TODO:

- [x] add a migration to add this field with the current default from metal-core to every super tenant network
- [ ] https://github.com/metal-stack/metalctl/pull/260

### CODE REVIEW

File: cmd/metal-api/internal/service/switch-service.go [hunk 1]
```diff
@@ -1026,7 +1044,7 @@ func findSwitchReferencedEntities(s *metal.Switch, ds *datastore.RethinkStore) (
 	return p, ips.ByProjectID(), m, ss, nil
 }
 
-func makeSwitchResponseList(ss metal.Switches, ds *datastore.RethinkStore) ([]*v1.SwitchResponse, error) {
+func (r *switchResource) makeSwitchResponseList(ss metal.Switches, ds *datastore.RethinkStore) ([]*v1.SwitchResponse, error) {
```
Comment (vknabel) [inline cmd/metal-api/internal/service/switch-service.go:1047]: for `makeSwitchResponse` you dropped the datastore parameter. Is there a reason not to do the same for `makeSwitchResponseList`?

Comment (majst01) [reply cmd/metal-api/internal/service/switch-service.go:1047]: Ah thanks, one moment

Comment (majst01) [reply cmd/metal-api/internal/service/switch-service.go]: fixed

**Review by vknabel — APPROVED** [review]

File: cmd/metal-api/internal/service/network-service.go [hunk 2]
```diff
@@ -409,6 +417,23 @@ func (r *networkResource) createNetwork(request *restful.Request, response *rest
...
+			return nil, errors.New("additionalannouncablecidrs can only be set in a private super network")
+		}
+		for _, cidr := range additionalCidrs {
+			_, err := netip.ParsePrefix(cidr)
+			if err != nil {
+				return nil, fmt.Errorf("given cidr:%q in additionalannouncablecidrs is malformed:%w", cidr, err)
+			}
+			result = append(result, cidr)
+		}
+	}
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/network-service.go:433]: Indent level can be reduced:

```suggestion
	if len(additionalCidrs) == 0 {
	  return nil
	}
	
        if !privateSuper {
		return nil, errors.New("additionalannouncablecidrs can only be set in a private super network")
	}
	
	...
```

**Review by mwennrich — APPROVED** [review]

File: cmd/metal-api/internal/service/network-service.go (hunk 2 above)
Comment (majst01) [reply cmd/metal-api/internal/service/network-service.go]: yes, done

File: cmd/metal-api/internal/metal/network.go [hunk 3]
```diff
@@ -207,17 +207,18 @@ func (p *Prefix) equals(other *Prefix) bool {
...
+	PartitionID                string            `rethinkdb:"partitionid" json:"partitionid"`
+	ProjectID                  string            `rethinkdb:"projectid" json:"projectid"`
+	ParentNetworkID            string            `rethinkdb:"parentnetworkid" json:"parentnetworkid"`
+	Vrf                        uint              `rethinkdb:"vrf" json:"vrf"`
+	PrivateSuper               bool              `rethinkdb:"privatesuper" json:"privatesuper"`
+	Nat                        bool              `rethinkdb:"nat" json:"nat"`
+	Underlay                   bool              `rethinkdb:"underlay" json:"underlay"`
+	Shared                     bool              `rethinkdb:"shared" json:"shared"`
+	Labels                     map[string]string `rethinkdb:"labels" json:"labels"`
+	AdditionalAnnouncableCIDRs []string          `rethinkdb:"additionalannouncablecidrs" json:"additionalannouncablecidrs" description:"list of cidrs which are added to the route maps per tenant private network, these are typically pod- and service cidrs, can only be set in a supernetwork"`
```
Comment (Gerrit91) [inline cmd/metal-api/internal/metal/network.go:221]:
```suggestion
	AdditionalAnnouncableCIDRs []string          `rethinkdb:"additionalannouncablecidrs" json:"additionalannouncablecidrs" description:"list of cidrs which are added to the route maps per tenant private network, these are typically pod- and service cidrs, can only be set when private super is set to true"`
```

File: cmd/metal-api/internal/service/network-service.go [hunk 4]
```diff
@@ -409,6 +417,28 @@ func (r *networkResource) createNetwork(request *restful.Request, response *rest
 	r.send(request, response, http.StatusCreated, v1.NewNetworkResponse(nw, usage))
 }
 
+func validateAdditionalAnnouncableCIDRs(additionalCidrs []string, privateSuper bool) ([]string, error) {
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/network-service.go:420]: Maybe more like this?

```suggestion
func validateAdditionalAnnouncableCIDRs(additionalCidrs []string, privateSuper bool) error {
```

**Review by Gerrit91 — APPROVED** [review]

File: cmd/metal-api/internal/service/switch-service.go [hunk 5]
```diff
@@ -894,16 +911,16 @@ func makeBGPFilter(m metal.Machine, vrf string, ips metal.IPsMap) (v1.BGPFilter,
...
 	} else {
-		filter, err = makeBGPFilterMachine(m, ips)
+		filter, err = r.makeBGPFilterMachine(m, ips)
 	}
 
 	return filter, err
 }
 
-func makeSwitchNics(s *metal.Switch, ips metal.IPsMap, machines metal.Machines) (v1.SwitchNics, error) {
+func (r *switchResource) makeSwitchNics(s *metal.Switch, ips metal.IPsMap, machines metal.Machines) (v1.SwitchNics, error) {
```
[the remark this answers is not in the document]
Comment (majst01) [reply cmd/metal-api/internal/service/switch-service.go]: Also done

**Pull request merged**